src/main/java/uk/gov/hmcts/reform/managecase/zuulfilters/AuthHeaderRoutingFilter.java

```diff
@@ -0,0 +1,49 @@
+package uk.gov.hmcts.reform.managecase.zuulfilters;
+
+import com.netflix.zuul.ZuulFilter;
+import com.netflix.zuul.context.RequestContext;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.managecase.security.SecurityUtils;
+
+import static org.springframework.cloud.netflix.zuul.filters.support.FilterConstants.ROUTE_TYPE;
+import static org.springframework.cloud.netflix.zuul.filters.support.FilterConstants.SIMPLE_HOST_ROUTING_FILTER_ORDER;
+
+@Component
+public class AuthHeaderRoutingFilter extends ZuulFilter {
+    public static final String AUTHORIZATION = "Authorization";
+    public static final String SERVICE_AUTHORIZATION = "ServiceAuthorization";
+
+    @Autowired
+    private SecurityUtils securityUtils;
+
+    @Override
+    public String filterType() {
+        return ROUTE_TYPE;
+    }
+
+    @Override
+    public int filterOrder() {
+        return SIMPLE_HOST_ROUTING_FILTER_ORDER - 1;
+    }
+
+    @Override
+    public boolean shouldFilter() {
+        return RequestContext.getCurrentContext().getRouteHost() != null
+            && RequestContext.getCurrentContext().sendZuulResponse();
+    }
+
+    @Override
+    public Object run() {
+        RequestContext context = RequestContext.getCurrentContext();
```

Comment:
just to double check, are we doing the following checks somewhere:
validate the incoming call has got a valid IDAM and S2S token

Reply:
First bullet point is taken care by spring security. Second one I think needs implementing.

Reply:
Implemented second one as well.